Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -167,12 +169,19 @@ - (instancetype)initWithLocalStore:(FSTLocalStore *)localStore
     [_limboCollector addGarbageSource:_limboDocumentRefs];
 
     _mutationCompletionBlocks = [NSMutableDictionary dictionary];
-    _targetIdGenerator = [FSTTargetIDGenerator generatorForSyncEngineStartingAfterID:0];
+    _targetIdGenerator = new firebase::firestore::core::TargetIdGenerator(
```

Comment:
You don't have to do this manual memory management. Objective-C++ objects will call the destructors of members correctly without implementing `dealloc`. However, you must declare them as an actual member variable instead of a property for it to work.

There are already several examples in the leveldb bits, e.g.

https://github.com/firebase/firebase-ios-sdk/blob/821fb909a3007f697ad96d83c6b8e8cc70df0b6b/Firestore/Source/Local/FSTLevelDBMutationQueue.mm#L88

In this case I think it would be something like this:

```objc
@implementation FSTSyncEngine {
  TargetIdGenerator _targetIdGenerator;
}

- (instancetype)initWithLocalStore:(FSTLocalStore *)localStore
                       remoteStore:(FSTRemoteStore *)remoteStore
                       initialUser:(FSTUser *)initialUser {
  if (self = [super init]) {
    // ...
   _targetIdGenerator = TargetIdGenerator::SyncEngineTargetIdGenerator(0);
  }
  return self;
}
```

This way you don't have to call `new` and you don't have to implement `dealloc`.

Reply:
done